File: daa/lab2/toh.c
#include <stdio.h>
void toh(int n,char s,char i,char t)
{
	if(n>=1)
	{
		toh(n-1,s,t,i);
		printf("Move disc %d from %c to %c\n",n,s,t);
		toh(n-1,i,s,t);
	}
}
int main()
{
	int n;
	printf("Enter number of discs:");
	scanf("%d",&n);
	toh(n,'S','I','T');
	return 0;
}

File: daa/lab2/toh.py
def toh(n,s,i,t):
	if(n>=1):
		toh(n-1,s,t,i)
		print("Move disc",n,"from",s,"to",t)
		toh(n-1,i,s,t)
def main():
	n=int(input("Enter number of discs:"))
	toh(n,'S','I','T')
if __name__=='__main__':
	main()
